stacbuilder/builder.py

```diff
@@ -688,12 +705,17 @@ def _override_collection_components(self, data: Dict[str, Any]) -> None:
 
     @staticmethod
     def _validate_collection(collection: Collection):
-        """Run STAC validation on the collection."""
+        """Run STAC validation on the collection.
+        TODO: remove this function, it is not used.
```

Comment:
TODO is no longer correct, this method is clearly in use.
```suggestion
```

Reply:
Or is it? I think I may be wrong about that after all. Maybe check that, and remove it if it is not used.

Reply:
AssetMetadataPipe also has a function to validate a collection. (I think we should remove one of both)
Perhaps the right chose is to make this one static and public and replace the one from AssetPipeline. What do you think?